src/engine/PiecewiseLinearConstraint.h

```diff
@@ -254,6 +255,37 @@ class PiecewiseLinearConstraint : public ITableau::VariableWatcher
     */
     virtual void getCostFunctionComponent( Map<unsigned, double> &/* cost */ ) const {}
 
+    /*
+      Ask the piecewise linear constraint to add its cost term corresponding to
+      the given phase to the cost function.
+      Nothing should be added when the constraint is fixed or inactive.
+      Minimizing the added term should lead to the constraint being
+      "closer to satisfied" in the given phase status.
+    */
+    virtual void addCostFunctionComponent( Map<unsigned, double> &/* cost */,
+                                           PhaseStatus /* phase */ ) const {}
+
+    /*
+      Compute the error of the current assignment with respect to the cost compoenent
+      corresponding to the given phase.
+    */
+    virtual double computeCostFunctionComponent( PhaseStatus /* phase */ ) const
+    {
+        throw MarabouError( MarabouError::FEATURE_NOT_YET_SUPPORTED,
+                            "Sum-of-Infeasibilities for the current constraint type"
+                            " is not supported." );
+    }
+
+    /*
+      Ask the piecewise linear constraint to remove the cost term corresponding
+      to the given phase from the cost function.
+      The method does not check whether the cost term has been previously added.
+      This is done in the SumOfInfeasibilitiesManager class, which maintains/updates the
+      SoI function.
+    */
+    virtual void removeCostFunctionComponent( Map<unsigned, double> &/* cost */,
```

Comment:
Instead of add/remove, why not have a get(), and let the manager do with it as it will?

Reply:
Yeah that makes sense. I'll defer this method to the SoIManager class.